src/includes/load-service.h

```diff
@@ -1215,6 +1233,19 @@ static void value_var_subst(const char *setting_name, std::string &line,
     line = std::move(r_line);
 }
 
+// Reads a dependency name while performing minimal argument expansion in it.
+inline string read_dependency_value(const char *setting_name, file_pos_ref input_pos, string_iterator &i,
+        string_iterator end, std::list<std::pair<unsigned,unsigned>> &offsets, const char *argval)
+{
+    string rval;
+    if (!argval) argval = "";
```

Comment:
Does this prevent erroring out in case `$1` is used but no service argument has been supplied?

Reply:
that was a leftover from when i was doing the custom expansion and forgot to remove it, fixed now